src/Controller/SearchController.php

```diff
@@ -0,0 +1,145 @@
+<?php
+
+/*
+ * This file is part of the Sylius package.
+ *
+ * (c) Paweł Jędrzejewski
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Lakion\SyliusElasticSearchBundle\Controller;
+
+use FOS\RestBundle\View\ConfigurableViewHandlerInterface;
+use FOS\RestBundle\View\View;
+use Lakion\SyliusElasticSearchBundle\Form\Type\FilterSetType;
+use Lakion\SyliusElasticSearchBundle\Search\Criteria\Criteria;
+use Lakion\SyliusElasticSearchBundle\Search\SearchEngineInterface;
+use Symfony\Component\Form\FormFactoryInterface;
+use Symfony\Component\HttpFoundation\Request;
+use Symfony\Component\HttpFoundation\Response;
+use Symfony\Component\HttpKernel\Exception\HttpException;
+
+/**
+ * @author Arkadiusz Krakowiak <arkadiusz.krakowiak@lakion.com>
+ */
+final class SearchController
+{
+    /**
+     * @var ConfigurableViewHandlerInterface
+     */
+    private $restViewHandler;
+
+    /**
+     * @var SearchEngineInterface
+     */
+    private $searchEngine;
+
+    /**
+     * @var FormFactoryInterface
+     */
+    private $formFactory;
+
+    /**
+     * @param ConfigurableViewHandlerInterface $restViewHandler
+     * @param SearchEngineInterface $searchEngine
+     * @param FormFactoryInterface $formFactory
+     */
+    public function __construct(
+        ConfigurableViewHandlerInterface $restViewHandler,
+        SearchEngineInterface $searchEngine,
+        FormFactoryInterface $formFactory
+    ) {
+        $this->restViewHandler = $restViewHandler;
+        $this->searchEngine = $searchEngine;
+        $this->formFactory = $formFactory;
+    }
+
+    /**
+     * @param Request $request
+     *
+     * @return Response
+     */
+    public function filterAction(Request $request)
+    {
+        $view = View::create();
+        if ($this->isHtmlRequest($request)) {
+            $view->setTemplate($this->getTemplateFromRequest($request));
+        }
+
+        $form = $this->formFactory->create(FilterSetType::class, Criteria::fromQueryParameters($this->getResourceClassFromRequest($request), []), ['filter_set' => $this->getFilterScopeFromRequest($request)]);
+        $form->handleRequest($request);
+        $criteria = $form->getData();
+        $criteria = Criteria::fromQueryParameters($this->getResourceClassFromRequest($request), array_merge($request->query->all(), $request->attributes->all(), $criteria->getFiltering()->getFields()));
```

Comment:
This named constructor is confusing, because it clearly takes something more than query parameters, I had to dig much deeper than expected to figure out why it is this way. :) Do we need to class to be configured here at all? Check my comment below.